README.md

```diff
@@ -1,21 +1,96 @@
+Quickly and easily obtain an xbox token to authenticate with Minecraft/Mojang
 
-## Usage
+## Installation
+```shell
+npm install xboxlive-auth
+```
+
+## Getting A Minecraft Java Token
+
+### Device Code Authentication
+```js
+const { Authflow } = require('@prismarinejs/xboxlive-auth');
+
+const doAuth = async() => {
+    const flow = new Authflow('mineflayer@is.cool', './')
+    const XSTSToken = await flow.getMinecraftJavaToken()
+    console.log(XSTSToken)
+}
+
+doAuth()
+```
+
+### Password-based Authentication
+```js
+const { Authflow } = require('@prismarinejs/xboxlive-auth');
+
+const doAuth = async() => {
+    const flow = new Authflow('mineflayer@is.cool', './', { password: 'thisIsAFakePassword123'})
+    const XSTSToken = await flow.getMinecraftJavaToken()
+    console.log(XSTSToken)
+}
 
+doAuth()
+```
+
+## Getting A Minecraft Bedrock Token
+When authenticating for minecraft bedrock edition, you have to generate and send a signed certificate to the bedrock server.
+This function handles logging into Xbox Live, posting its public key to the Mojang API, and using the returned certificate signed by Mojang.
+
+### Device Code Authentication
 ```js
-const template = require('prismarine-template')
+const { Authflow } = require('@prismarinejs/xboxlive-auth');
+const crypto = require('crypto')
+const curve = 'secp384r1'
 
-template.helloWorld()
+const keypair = crypto.generateKeyPairSync('ec', { namedCurve: curve }).toString('base64') 
+const doAuth = async() => {
+    const flow = new Authflow('mineflayer@is.cool', './')
+    const XSTSToken = await flow.getMinecraftBedrockToken(keypair)
+    console.log(XSTSToken)
+}
+
+doAuth()
 ```
 
-## API
+### Password Authentication:
+```js
+const { Authflow } = require('@prismarinejs/xboxlive-auth');
+const crypto = require('crypto')
+const curve = 'secp384r1'
+
+const keypair = crypto.generateKeyPairSync('ec', { namedCurve: curve }).toString('base64') 
+
+const doAuth = async() => {
+    const flow = new Authflow('mineflayer@is.cool', './', { password: 'thisIsAFakePassword123'})
+    const XSTSToken = await flow.getMinecraftBedrockToken(keypair)
+    console.log(XSTSToken)
+}
 
-### helloWorld()
+doAuth()
+```
+
+### Expected Response
+```php
+{
+    "userXUID": "2584878536129841", // May be null
+    "userHash": "3218841136841218711",
+    "XSTSToken": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJuYW1lIjoiWGJveFJlcGxheS5uZXQifQ.c2UraxPmZ4STYozrjFEW8SBqU0WjnIV0h-jjnfsKtrA",
+    "expiresOn": "2020-04-13T05:43:32.6275675Z"
+}
+```
 
-Prints hello world
+### Parameters
+Authflow class
+- username {String} - Username for authentication
+- cacheDirectory {String} - Where we will store your tokens
+- options {Object?}
+    - [password] {string} - If passed we will do password based authentication.
+    - [authTitle] {string} - Used in authenticating with Switch or Bedrock accounts. See https://github.com/PrismarineJS/xboxlive-auth/src/Constants.js
```

Comment:
this is not related to bedrock, this should always be set if possible. It is required to switch to live.com auth and do title authentication which is needed for accounts with a date of birth <18 years old. You can update all of the examples to set this.